src/CMakeLists.txt

```diff
@@ -392,36 +392,39 @@ if (ENABLE_BLACKMAGIC)
 endif()
 
 ################## OPENCV ###################
-
-find_package( OpenCV 4 )
-if (OpenCV_FOUND)
-  message("\nCOMPILING WITH OPENCV\n")
-  list(APPEND CMAKE_SWIG_FLAGS -DUSE_OPENCV=1)
-  target_compile_definitions(openshot PUBLIC USE_OPENCV=1)
-else()
-  message("\nOPENCV NOT FOUND, SOME FUNCTIONALITIES WILL BE DISABLED\n")
-endif()
-
-################## PROTOBUF ##################
-if (OpenCV_FOUND)
-  find_package(Protobuf 3)
-  
-  if (NOT Protobuf_FOUND)
-    # Protobuf is required when compiling with opencv
-    message(FATAL_ERROR "\nPLEASE INSTALL PROTOBUF. Protobuf is required when compiling with opencv.\n")
+if(ENABLE_OPENCV)
+  find_package(OpenCV 4)
+  if(NOT OpenCV_FOUND)
+    set(ENABLE_OPENCV FALSE CACHE BOOL
+      "Build with OpenCV algorithms (requires Boost, Protobuf 3)" FORCE)
```

Comment:
Basically, `OpenCV` isn't discovered with `REQUIRED` because it's _not_, and if `ENABLE_OPENCV` is `ON` (by default or explicitly) but it's not found, the option will just forcibly reset itself to `OFF`. Any subsequent consumption of any of the OpenCV pieces is then made conditional on `ENABLE_OPENCV` being a truthy value.